StoreApp/Controllers/StoreController.cs

```diff
@@ -39,6 +43,34 @@ public ActionResult Index()
             }
         }
 
+        [HttpPost]
+        public ActionResult Index(FilterProductsViewModel filter)
+        {
+            TempData["Filter"] = filter;
+
+            if (!ModelState.IsValid)
+            {
+                var products = productService.GetAll().Select(p => webMapper.Map(p));
+
+                return View(products);
+            }
+            
+            FilterProductsDTO filterDTO = webMapper.config.Map<FilterProductsViewModel, FilterProductsDTO>(filter);
+            
+            var productsFiltered = filterProductsService.GetProductsDTOFiltered(filterDTO).Select(p => webMapper.Map(p));
+
+            return View(productsFiltered);
+        }
+
+        public ActionResult FilterMenuPartial()
+        {
+            TempData["Categories"] = new SelectList(categoryService.GetAll(), dataValueField: "Id", dataTextField: "Name");
+            TempData["Brands"] = new SelectList(brandService.GetAll(), dataValueField: "Id", dataTextField: "Name");
+            TempData["Producers"] = new SelectList(producerService.GetAll(), dataValueField: "Id", dataTextField: "Name");
```

Comment:
As I can see we use this block of code a lot of times. So I propose to optimize code and implement the principle DRY.